docs/synapse/userguides/storm_ref_filter.rstorm

```diff
@@ -907,41 +969,119 @@ Refer to the :ref:`storm-ref-subquery` guide for additional information on subqu
 
+- From an initial set of IPv4 addresses, filter results to only those IPv4s registered to an Autonomous
+  System (AS) whose name starts with "makonix":
 
 .. storm-pre:: inet:ipv4 +{ :asn -> inet:asn +:name^="makonix" }
 ::
 
     <inet:ipv4> +{ :asn -> inet:asn +:name^="makonix" }
 
 
-- From an initial set of ``file:bytes`` nodes, return only those that are detected as malicious by 10 or more AV vendors (i.e., that are associated with 10 or more ``it:av:filehit`` nodes):
-
+- From an initial set of ``file:bytes`` nodes, filter results to only those that are detected as
+  malicious by ten (10) or more antivirus / malscanner vendors (i.e., files that are associated with
+  10 or more ``it:av:filehit`` nodes):
 
 .. storm-pre:: file:bytes +{ -> it:av:filehit }>=10
 ::
 
     <file:bytes> +{ -> it:av:filehit }>=10
 
+- From an initial set of x509 certificates (``crypto:x509:cert``), filter results to only those
+  certificates linked to more than one FQDN (``inet:fqdn``) identity:
+
+.. storm-pre:: [ ( crypto:x509:cert=* :identities:fqdns=(woot.com,) ) ( crypto:x509:cert=* :identities:fqdns=(hurr.net, derp.org) ) ] +{ :identities:fqdns -> inet:fqdn }>1
+::
+  
+  <crypto:x509:cert> +{ :identities:fqdns -> inet:fqdn }>1
+
+
+.. _filter-expression:
+
+Expression Filters
+------------------
+
+You can filter your current set of data (nodes) based on the evaluation of a particular expression.
+Expression filters are useful when you need to compute a value that you want to use for the filter,
+or when you want to filter based on a value that may change (e.g., when using Storm queries that
+assign variables - see :ref:`storm-adv-vars`).
+
+**Syntax:**
+
+*<query>* **+** | **-** **$(** *<expression>* **)**
+
+
+**Examples:**
+
+- From an initial set of network flows (``inet:flow`` nodes), filter results to only those flows where
+  the total number of bytes transferred in the flow between the source (``inet:flow:src:txbytes``) and
+  destination (``inet:flow:dst:txbytes``) is greater than 100MB (~100,000,000 bytes):
+
+.. storm-pre:: [ ( inet:flow=* :src:txbytes=60000000 :dst:txbytes=60000000 ) ( inet:flow=* :src:txbytes=1024 :dst:txbytes=4096 ) ] +$( :src:txbytes + :dst:txbytes >=100000000 )
+::
+  
+  <inet:flow> +$( :src:txbytes + :dst:txbytes >=100000000 )
+
+
+- From an initial set of x509 certificates (``crypto:x509:cert``), filter results to only those
+  certificates linked to more than one FQDN (``inet:fqdn``) identity:
+
+.. storm-pre:: [ ( crypto:x509:cert=* :identities:fqdns=(woot.com,) ) ( crypto:x509:cert=* :identities:fqdns=(hurr.net, derp.org) ) ] $fqdns=:identities:fqdns +$( $fqdns.size() > 1 )
+::
+  
+  <crypto:x509:cert> $fqdns=:identities:fqdns +$( $fqdns.size() > 1 )
+
+This example assigns the set of domains in the ``crypto:x509:cert:identities:fqdns`` property to the
+user-defined variable ``$fqdns``, computes the number of domains in the set using :ref:`stormprims-set-size`,
```

Comment:
It is actually a List prim, not a Set, https://synapse.docs.vertex.link/en/latest/synapse/autodocs/stormtypes_prims.html#size